skill-tree/src/main/java/com/RDS/skilltree/exceptions/GlobalExceptionHandler.java

```diff
@@ -135,4 +135,12 @@ public ResponseEntity<?> handleForbiddenException(ForbiddenException ex, WebRequ
         log.error("Exception - Error : {}", ex.getMessage(), ex);
         return new ResponseEntity<>(new GenericResponse<>(null, ex.getMessage()), HttpStatus.FORBIDDEN);
     }
+
+    @ExceptionHandler(InternalServerErrorException.class)
```

Comment:
The exception object ex is logged, but the response only includes the error message. Consider including more details about the exception in the response for debugging purposes, especially in non-production environments. However, be mindful of exposing sensitive information.

```
@ExceptionHandler(InternalServerErrorException.class)
public ResponseEntity<?> handleInternalServerErrorException(
        InternalServerErrorException ex, WebRequest request) {
    log.error("Internal Server Error", ex); // Log without sensitive information

    // Create a more specific error message based on the exception type or cause
    String errorMessage = "An unexpected error occurred.";

    // Consider adding more details to the response for debugging
    Map<String, Object> errorDetails = new HashMap<>();
    errorDetails.put("timestamp", LocalDateTime.now());
    errorDetails.put("message", errorMessage);
    errorDetails.put("details", ex.getMessage()); // Include exception details for debugging

    return new ResponseEntity<>(errorDetails, HttpStatus.INTERNAL_SERVER_ERROR);
}

```

Reply:
- made the change